src/arrayMethodSort.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    compareFunction = compareFunction || function(a,b){
+      const aString = String(a);
+      const bString = String(b);
+
+      if(aString > bString) return 1;
+      if(aString < bString) return -1;
+      return 0;
+    };
+
+    for(let i = 0; i < this.length - 1; i++){
```

Comment:
The outer loop runs one time too many. Since the inner loop already handles the comparisons, you can end the outer loop at `this.length - 1`. This is a small optimization to avoid unnecessary iterations.